### ISSUE TITLE
Post API using Form Data showing ugly namespace Body_route_def_post in Schemas section

### ISSUE THREAD

**Issue by devidkarundeng** [tiangolo/fastapi#2754]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

---

I am implementing a simple OAuth POST API using Form Body, with client_id, client_secret. Everything is good, this is purely cosmetic annoyance seeing the Schemas at the bottom of Open API with ugly namespace. Is there a way to polish the naming of "Body_oauth_authorize_oauth_authorize_post" (Body_route_def_post)? 

I have tried to define name using Pydantic BaseModel, and have the body form to depends on the Pydantic class, but still the namespaces in the Open API Schemas showing "Body_oauth_authorize_oauth_authorize_post" (Body_route_def_post).

`formbody: NewName = Depends(NewName.as_form)`

<img width="520" alt="Screen Shot 2021-02-04 at 7 22 55 PM" src="https://user-images.githubusercontent.com/40422546/106892360-8ac86200-671e-11eb-8d7a-8b2e4e405990.png">

**Comment by ycd** [comment]
You can add title to class [config](https://pydantic-docs.helpmanual.io/usage/model_config/).

```python

class SomePydanticModel(BaseModel):
     some_field: str
     
     class Config:
         title = "CustomTitle"
```

**Comment by devidkarundeng** [comment]
class Config Custom Title doesn't seem to work when converted into Form. It works fine if not as form. All the POST Form type has a weird namespaces in the Schema section.. Code and screenshot of the Schema below.

<img width="495" alt="Screen Shot 2021-02-09 at 11 54 34 AM" src="https://user-images.githubusercontent.com/40422546/107317760-d76cbe00-6acd-11eb-8531-bf84dc9295db.png">


```Python
class OAuthAccessToken(BaseModel):
    client_id: str
    client_secret: str
    
    class Config:
        title = "OAuthAccessToken"
    
    @classmethod
    def as_form(
        cls,
        client_id: str = Form(...),
        client_secret: str = Form(...),
    ):
        return cls(client_id, client_secret)


@router.post(
    "/oauth/access-token", 
    summary="Get user's access token",
) 
async def oauth_access_token(
    form_data: OAuthAccessToken = Depends(OAuthAccessToken.as_form)
):
```

**Comment by ycd** [comment]
Try this:


```python
def form_body(cls):
    cls.__signature__ = cls.__signature__.replace(
        parameters=[
            arg.replace(default=Form(...))
            for arg in cls.__signature__.parameters.values()
        ]
    )
    return cls


@form_body
class LoginForm(BaseModel):
    client_id: str
    client_secret: str

    class Config:
        title = "LoginForm"


@app.post(
    "/oauth/access-token",
)
async def oauth_access_token(form_data: LoginForm):
    ...
```

**Comment by devidkarundeng** [comment]
Thanks for the tip @ycd! It WORKS!

**Comment by ycd** [comment]
You are welcome! Thanks for closing the issue.

**Comment by devidkarundeng** [comment]
Actually I lied.. Got too excited and didn't check thoroughly.. it doesn't turn the params into Form Body.. still JSON body...

**Comment by ycd** [comment]
Did you tried adding depends?

```python
@app.post(
    "/oauth/access-token",
)
async def oauth_access_token(form_data: LoginForm = Depends(LoginForm)):
```

**Comment by devidkarundeng** [comment]
Yes, adding depends will turn it into Form Body, but the Schemas Model at the bottom will look funky again..

**Comment by ycd** [comment]
I traced this down:

https://github.com/tiangolo/fastapi/blob/33be5fc8baa02bae949e1986f1821dae8f84f487/fastapi/dependencies/utils.py#L169-L201

This function assumes dependant is Callable but not an instance of BaseModel, so if dependant is BaseModel it's config gets overriden. so i added this. 

This adds configs title to the currently flatted dependant.

```python
def get_flat_dependant(
    dependant: Dependant,
    *,
    skip_repeats: bool = False,
    visited: Optional[List[CacheKey]] = None,
) -> Dependant:
    if visited is None:
        visited = []
    visited.append(dependant.cache_key)
    
    if dependant.call is not None:
        if hasattr(dependant.call, "Config"):
            config = getattr(dependant.call, "Config")
            if hasattr(config, "title"):
                dependant.name = config.title   

    flat_dependant = Dependant(
        ...
        name=dependant.name or None     
    )
```

However, this is not the best implementation idea, imho, i'll take another look at this later.

**Comment by devidkarundeng** [comment]
@ycd Thanks a lot for looking into this despite just cosmetic annoyance issue! Will try your tips later.

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for reporting back and closing the issue @devidkarundeng  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
